### PR TITLE
Update plugins

### PR DESCRIPTION
**Pull request by batmat**
Round of plugins update based on `versions:display-plugins-updates`, similar to https://github.com/jenkinsci/pom/pull/37

I left aside:
* `frontend-maven-plugin` handled in a dedicated PR (https://github.com/jenkinsci/plugin-pom/pull/138).
* `access-modifier` because it creates issues with RequireUpperBoundDeps which I'll probably handle separately to keep this PR more straightforward, given there's already enough things touched here.

* [x] :heavy_check_mark: Successful PR build of this on the `git-plugin` (using `3.29-20181202.133920-6`): https://github.com/jenkinsci/git-plugin/pull/641

Output was:

```
[INFO] The following plugin updates are available:
[INFO]   com.github.eirslett:frontend-maven-plugin .............. 1.4 -> 1.6
[INFO]   maven-clean-plugin ................................. 3.0.0 -> 3.1.0
[INFO]   maven-compiler-plugin .............................. 3.6.1 -> 3.8.0
[INFO]   maven-dependency-plugin ............................ 3.0.0 -> 3.1.1
[INFO]   maven-deploy-plugin ............................. 2.8.2 -> 3.0.0-M1
[INFO]   maven-enforcer-plugin ........................ 3.0.0-M1 -> 3.0.0-M2
[INFO]   maven-failsafe-plugin .......................... 2.19.1 -> 3.0.0-M1
[INFO]   maven-install-plugin ............................ 2.5.2 -> 3.0.0-M1
[INFO]   maven-invoker-plugin ............................... 2.0.0 -> 3.1.0
[INFO]   maven-jar-plugin ................................... 3.0.2 -> 3.1.0
[INFO]   maven-javadoc-plugin .............................. 2.10.4 -> 3.0.1
[INFO]   maven-resources-plugin ............................. 3.0.2 -> 3.1.0
[INFO]   maven-scm-plugin .................................. 1.9.5 -> 1.11.1
[INFO]   maven-surefire-plugin ............................ 2.20 -> 3.0.0-M1
[INFO]   maven-war-plugin ................................... 3.0.0 -> 3.2.2
[INFO]   org.codehaus.mojo:mrm-maven-plugin ................. 1.0.1 -> 1.1.0
[INFO]   org.jacoco:jacoco-maven-plugin ..................... 0.7.9 -> 0.8.1
[INFO]   org.kohsuke:access-modifier-checker .................. 1.15 -> 1.16
```

### CODE REVIEW

**Comment by batmat** [comment]
@oleg-nenashev I reverted surefire and failsafe updates.

**Comment by batmat** [comment]
`3.29-20181203.165022-9` is the new timestamped SNAPSHOT. Updating the related https://github.com/jenkinsci/git-plugin/pull/641 test PR.

**Comment by batmat** [comment]
Git-plugin PR is successful https://github.com/jenkinsci/git-plugin/pull/641 FTR.

File: pom.xml [hunk 1]
```diff
@@ -302,23 +302,23 @@
       <plugins>
         <plugin>
           <artifactId>maven-clean-plugin</artifactId>
-          <version>3.0.0</version>
+          <version>3.1.0</version>
         </plugin>
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
-          <version>3.6.1</version>
+          <version>3.8.0</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Generally needed for support of higher Java levels. Also updated in #133

File: pom.xml [hunk 2]
```diff
@@ -330,19 +330,19 @@
         </plugin>
         <plugin>
           <artifactId>maven-deploy-plugin</artifactId>
-          <version>2.8.2</version>
+          <version>3.0.0-M1</version>
```
Comment (oleg-nenashev) [inline pom.xml]: https://issues.apache.org/jira/browse/MDEPLOY-244 as @jtnord commented in https://github.com/jenkinsci/pom/pull/37

**Comment by batmat** [comment]
@oleg-nenashev reverted the m-deploy-p bump with a comment. Care to have another look? Thanks

**Comment by oleg-nenashev** [comment]
It also fixes https://issues.apache.org/jira/browse/MDEP-571

**Pull request merged**